app/src/main/java/com/android/streetworkapp/ui/event/AddEvent.kt

```diff
@@ -103,57 +133,115 @@ fun AddEventScreen(
             }
-          },
-          modifier =
-              Modifier.align(Alignment.Center)
-                  .offset(0.dp, 100.dp)
-                  .padding(40.dp)
-                  .size(width = 150.dp, height = 40.dp)
-                  .testTag("addEventButton"),
-          containerColor = ColorPalette.INTERACTION_COLOR_DARK,
-          contentColor = ColorPalette.PRIMARY_TEXT_COLOR) {
-            Text("Add new event")
+            // for non users-related-input errors, we only show the message for a brief amount of
+            // time
+            LaunchedEffect(isTextEvaluationError) {
+              delay(AddEventParams.TEXT_EVALUATION_DISPLAY_TIME)
+              isTextEvaluationError = false
+            }
+
+            Button(
+                colors =
+                    ButtonDefaults.buttonColors(
+                        containerColor = ColorPalette.INTERACTION_COLOR_DARK,
+                        contentColor = ColorPalette.PRIMARY_TEXT_COLOR),
+                modifier = Modifier.testTag("addEventButton"),
+                onClick = {
+                  if (event.date.seconds <
+                      Timestamp(Calendar.getInstance(TimeZone.getDefault()).time).seconds) {
+                    isDateBackInTimeError.value = true
+                    formErrorMessage = AddEventFormErrorMessages.DATE_BACK_IN_TIME_ERROR
+                  } else if (event.title.isEmpty()) {
+                    isTitleEmptyError.value = true
+                    formErrorMessage = AddEventFormErrorMessages.TITLE_EMPTY_ERROR_MESSAGE
+                  } else {
+                    // making sure the description & title are valid according to our evaluation
+                    // thresholds
+                    val formattedTextInput =
+                        "Title: ${event.title}. Description: ${event.description}" // we format it
+                    // to do only one
+                    // api call
+                    textModerationViewModel.analyzeText(
+                        formattedTextInput,
+                        { isTextUnderThresholds ->
+                          if (isTextUnderThresholds) {
+                            createEvent(
+                                event,
+                                navigationActions,
+                                eventViewModel,
+                                userViewModel,
+                                parkViewModel,
+                                context)
+                          } else {
+                            isTextEvaluationOverThresholdsError.value = true
+                            formErrorMessage =
+                                AddEventFormErrorMessages.TEXT_EVALUATION_OVER_THRESHOLDS_ERROR
+                          }
+                        },
+                        {
+                          isTextEvaluationError = true
+                          formErrorMessage = AddEventFormErrorMessages.TEXT_EVALUATION_ERROR
+                        })
```

Comment:
For the same reasons as the above suggestion, i would suggest separating this code block (from 214 to 240) into a separate function to reduce complexity and avoid nested ifs, best way i managed to achieve this is by changing the definition of `formErrorMessage` to make it a `MutableState`:
`val formErrorMessage = remember { mutableStateOf("") }`

and modifying line 213 up to 241 like so:
```
if (formErrorMessage.value.isEmpty()) {
                  val error =
                      analyzeTextAndCreateEvent(
                          event,
                          navigationActions,
                          eventViewModel,
                          userViewModel,
                          parkViewModel,
                          context,
                          isTextEvaluationOverThresholdsError,
                          textModerationViewModel,
                          formErrorMessage)
                  isTextEvaluationError = error
                }
```

and create this private function:
```
private fun analyzeTextAndCreateEvent(
    event: Event,
    navigationActions: NavigationActions,
    eventViewModel: EventViewModel,
    userViewModel: UserViewModel,
    parkViewModel: ParkViewModel,
    context: Context,
    isTextEvaluationOverThresholdsError: MutableState<Boolean>,
    textModerationViewModel: TextModerationViewModel,
    formErrorMessage: MutableState<String>
): Boolean {
  val formattedTextInput = "Title: ${event.title}. Description: ${event.description}"
  var isTextEvaluationError = false
  textModerationViewModel.analyzeText(
      formattedTextInput,
      { isTextUnderThresholds ->
        if (isTextUnderThresholds) {
          createEvent(
              event, navigationActions, eventViewModel, userViewModel, parkViewModel, context)
        } else {
          isTextEvaluationOverThresholdsError.value = true
          formErrorMessage.value = AddEventFormErrorMessages.TEXT_EVALUATION_OVER_THRESHOLDS_ERROR
        }
      },
      {
        isTextEvaluationError = true
        formErrorMessage.value = AddEventFormErrorMessages.TEXT_EVALUATION_ERROR
      })

  return isTextEvaluationError
}
```

Reply:
I refactored the code in a function called `onAddEventClickHandler`

Reply:
Nice! this seems to have greatly reduced the cognitive complexity, thank you for the changes